### PR TITLE
[Seedless-Onboarding] export account

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
Button to export private key of a signer account.

## How to test it
Navigate to `Settings->Signer account` 
Click export

## Screenshots
Screenshot 2023-10-19 at 17 44 35
Screenshot 2023-10-19 at 17 44 41
Screenshot 2023-10-19 at 17 46 13

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/settings/ExportMPCAccount/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+const ExportMPCAccount = () => {
+  const mpcCoreKit = useMPC()
+
+  const [pk, setPk] = useState<string | undefined>()
+
+  const isLoggedIn = mpcCoreKit?.status === COREKIT_STATUS.LOGGED_IN
+
+  const exportPK = async () => {
+    const exportedPK = await mpcCoreKit?._UNSAFE_exportTssKey()
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/index.tsx:15]: This call seems to take a few seconds for me. I would suggest adding a loading state and disable the Export button to give the user some indication.

File: src/components/settings/ExportMPCAccount/index.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  }
+  return (
+    <Box>
+      <Box>
+        <Typography>
+          This action reveals the seedphrase / private key of your logged in signer account. This export can be used to
+          move the account into a self-custodial wallet application.
+        </Typography>
+        <Alert severity="warning" sx={{ mt: 3 }}>
+          Anyone who has access to this seedphrase / key has <b>full access</b> over your Signer Account. You should
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/index.tsx:34]:
```suggestion
          Anyone who has access to this seed phrase / key has <b>full access</b> over your Signer Account. You should
```

**Review by usame-algan — APPROVED** [review]

File: src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+type ExportFormData = {
+  [ExportFieldNames.password]: string
+}
+
+const ExportMPCAccountModal = ({ onClose, open }: { onClose: () => void; open: boolean }) => {
+  const { exportPk } = useContext(MpcWalletContext)
+
+  const [pk, setPk] = useState<string | undefined>()
+  const [isExporting, setIsExporting] = useState(false)
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx:22]: I think you can use the isSubmitting flag from RHF instead

File: src/hooks/wallets/mpc/useMPCWallet.ts [hunk 4]
```diff
@@ -125,12 +126,31 @@ export const useMPCWallet = (): MPCWalletHook => {
...
+
+    try {
+      if (securityQuestions.isEnabled()) {
+        // Only export PK if recovery works
+        await securityQuestions.recoverWithPassword(password)
+      }
+      const exportedPK = await mpcCoreKit?._UNSAFE_exportTssKey()
+      return exportedPK
+    } catch (err) {
+      // TODO: Throw error through sentry
```
Comment (usame-algan) [inline src/hooks/wallets/mpc/useMPCWallet.ts:143]: Related to my other comment, should we rethrow an error here in order to display it to the user in the UI?

File: src/components/settings/SignerAccountMFA/useMFASettings.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import useMPC from '@/hooks/wallets/mpc/useMPC'
+import { COREKIT_STATUS } from '@web3auth/mpc-core-kit'
+
+export type MFASettings = {
+  mfaEnabled: boolean
+} | null
+
+const useMFASettings = () => {
```
Comment (usame-algan) [inline src/components/settings/SignerAccountMFA/useMFASettings.ts:8]: I couldn't find where this hook is being used. What is the purpose of it?

Comment (schmanu) [reply src/components/settings/SignerAccountMFA/useMFASettings.ts:8]: Oh I think it's some legacy hook. I will merge the current epic branch in. That should remove this hook.

File: src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+type ExportFormData = {
+  [ExportFieldNames.password]: string
+  [ExportFieldNames.pk]: string | undefined
+}
+
+const ExportMPCAccountModal = ({ onClose, open }: { onClose: () => void; open: boolean }) => {
+  const { exportPk } = useContext(MpcWalletContext)
+  const [error, setError] = useState<string>()
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx:25]: nit: Just found out we can even use RHF to display an error separate from its input https://react-hook-form.com/docs/useformstate/errormessage

File: src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  const exportedKey = watch(ExportFieldNames.pk)
+
+  const onSubmit = async (data: ExportFormData) => {
+    try {
+      setError(undefined)
+      const pk = await exportPk(data[ExportFieldNames.password])
+      setValue(ExportFieldNames.pk, pk)
+    } catch (err) {
+      logError(ErrorCodes._305, err)
+      setError('Error exporting account. Your entered password might be invalid.')
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx:45]: Is it possible to reuse the same string that we throw from `exportPk`? I would slightly reword it to sound more active e.g. `Error exporting account. Make sure the password is correct.`

File: src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                  fullWidth
+                  error={!!formState.errors[ExportFieldNames.password]}
+                  helperText={formState.errors[ExportFieldNames.password]?.message}
+                  {...register(ExportFieldNames.password, {
+                    required: true,
+                  })}
+                />
+              </>
+            )}
+            {error && <ErrorMessage className={css.modalError}>{error}</ErrorMessage>}
```
Comment (usame-algan) [inline src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx:107]: I think the `ErrorMessage` component adds vertical margin by default which makes the UI jump a bit more. I would suggest removing it via the css class.

**Review by usame-algan — APPROVED** [review]
Looks good 👍

File: src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx (hunk 6 above)
Comment (usame-algan) [reply src/components/settings/ExportMPCAccount/ExportMPCAccountModal.tsx:25]: Although on second thought if we set the error via RHF it will not be persisted afair so this is probably better.
